src/app/components/payment-dialog/payment-dialog.component.ts

```diff
@@ -75,7 +89,70 @@ export class PaymentDialogComponent implements OnInit {
                     : undefined
         };
 
-        this.dialogRef.close(payload);
+        this.pendingTransferPollingService.pendingTransfers$
+            .pipe(
+                first(),
+                switchMap(pendingTransfers => {
+                    const pendingPayment = pendingTransfers.find(
+                        pendingTransfer =>
+                            pendingTransfer.token_address ===
+                                payload.tokenAddress &&
+                            pendingTransfer.role === 'initiator' &&
+                            pendingTransfer.locked_amount.isEqualTo(
+                                payload.amount
+                            ) &&
+                            pendingTransfer.target === payload.targetAddress &&
+                            (payload.paymentIdentifier
+                                ? pendingTransfer.payment_identifier.isEqualTo(
+                                      payload.paymentIdentifier
+                                  )
+                                : true)
+                    );
+
+                    if (pendingPayment) {
+                        const token = this.raidenService.getUserToken(
+                            payload.tokenAddress
+                        );
+                        const formattedAmount = amountToDecimal(
+                            payload.amount,
+                            payload.decimals
+                        ).toFixed();
+                        const confirmationPayload: ConfirmationDialogPayload = {
+                            title: 'Retrying payment',
+                            message:
+                                `There is already a payment of <b>${formattedAmount} ${
+                                    token.symbol
+                                }</b> being sent to <b>${
+                                    payload.targetAddress
+                                }</b>. <br/>` +
+                                'Are you sure you want to send the same payment again?'
```

Comment:
> payment

In the light client we were told to not use the word `payment` and instead use `transfer`. Not sure if that's also the case in the WebUI project, but I would think so?

Reply:
I see the point but we use the word `payment` everywhere. Also on the client itself and the docs. I would leave it here for a later renaming of the whole WebUI.